File: CabManagementSystemWeb/Entities/Employee.cs
using System.ComponentModel.DataAnnotations.Schema;
using CabManagementSystemWeb.Data;
using CabManagementSystemWeb.Dtos;

namespace CabManagementSystemWeb.Entities;

[Table("Employees")]
public class Employee : IEntity
{
    public int Id { get; set; }
    public User? User { get; set; }
    public required int UserId { get; set; }
    public Branch? Branch { get; set; }
    public required int BranchId { get; set; }
    public DateTime Created { get; set; }
    public DateTime? Updated { get; set; } = null;

    public EmployeeDetailDto ConvertToDetailDto()
    {
        return new EmployeeDetailDto() {
            Id = Id,
            UserId = UserId,
            BranchId = BranchId
        };
    }
}

File: cab-managemeny-system-spa/src/modules/employee/Employee.ts
export default class Employee {
    public id: number;
    public userId: number;
    public branchId: number;

    constructor(
        id: number,
        userId: number,
        branchId: number
    ) {
        this.id = id;
        this.userId = userId;
        this.branchId = branchId;
    }
}
